src/vs/editor/common/languages.ts

```diff
@@ -965,6 +965,17 @@ export interface DocumentHighlightProvider {
 	provideDocumentHighlights(model: model.ITextModel, position: Position, token: CancellationToken): ProviderResult<DocumentHighlight[]>;
 }
 
+export interface MultiDocumentHighlightProvider {
+	/**
+	 * Provide a Map of URI --> document highlights, like all occurrences of a variable or
+	 * all exit-points of a function.
+	 *
+	 * Used in cases such as split view, notebooks, etc. where there can be multiple documents
+	 * with shared symbols.
+	 */
+	provideMultiDocumentHighlights(primaryModel: model.ITextModel, position: Position, wordSeparators: string, token: CancellationToken, otherModels: model.ITextModel[]): ProviderResult<Map<URI, DocumentHighlight[]>>;
```

Comment:
- style `token` should come last, also for MT/EH-IPC calls only tokens that are the last position are considered
- I would assume that `wordSeparators` is an implementation detail of one particular provider and nothing generic that should make it into the API